### PR TITLE
Refactor web configuration/ Add dartdevc

### PR DESCRIPTION
**Pull request by jonahwilliams**
Place all of the flutter web engine stuff its own rule/directory.
Via dart2js:

 ~/Documents/engine/src/out/host_debug_unopt/dart-sdk/bin/dart2js -o ext.js --libraries-spec=/Users/jonahwilliams/Documents/engine/src/out/host_debug_unopt/flutter_patched_sdk/lib/libraries.json  lib/main.dart

Via dartdevc:

~/Documents/engine/src/out/host_debug_unopt/dart-sdk/bin/dartdevc -k lib/main.dart -o ext.js --dart-sdk-summary=/Users/jonahwilliams/Documents/engine/src/out/host_debug_unopt/flutter_patched_sdk/lib/_internal/ddc_sdk.dill

This will break the current dart2js compile test, so I will disable that in the meantime

### CODE REVIEW

**Comment by jonahwilliams** [comment]
These still need the correctly built platform dill files for dartdevc and dart2js, as well as the browser files for dartdevc.

**Comment by jonahwilliams** [comment]
cc @Hixie for licensing question: the libraries.dart file is only needed to execute a script that produces the platform dill files and dartdevc sdk (precompiled dart files). Does this have a different licensing requirement?

**Comment by jonahwilliams** [comment]
I'm going to leave out the platform dill generation for now then

File: web_sdk/libraries.json [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        "uri": "lib/isolate/isolate.dart"
+      },
+      "developer": {
+        "patches": "lib/_internal/js_runtime/lib/developer_patch.dart",
+        "uri": "lib/developer/developer.dart"
+      },
+      "web_gl": {
+        "uri": "lib/web_gl/dart2js/web_gl_dart2js.dart"
+      },
+      "indexed_db": {
```
Comment (yjbanov) [inline web_sdk/libraries.json]: Curious what would happen if we removed some of the libraries we know we're not using.

**Comment by jonahwilliams** [comment]
The build release test passed on cirrus, but the github status did not update

**Pull request merged**